bin/run.sh

```diff
@@ -20,14 +20,19 @@ case $1 in
     cat version.json
     ;;
   test)
-    python3 -m venv /tmp/tests
-    /tmp/tests/bin/pip install -e jobs/
-    /tmp/tests/bin/pip install -r jobs/requirements/default.txt -r jobs/requirements/dev.txt -c jobs/requirements/constraints.txt
-    /tmp/tests/bin/py.test --ignore=jobs/tests/test_lamdba_s3_event_functional.py --override-ini="cache_dir=/tmp/tests" jobs/tests
+    py.test --ignore=jobs/tests/test_lamdba_s3_event_functional.py --override-ini="cache_dir=/tmp/tests" jobs/tests
     ;;
   lambda.zip)
-    cd .venv/lib/python3.6/site-packages/
+    pip install virtualenv
+    rm -fr .venv
+    virtualenv .venv
+    source .venv/bin/activate
+    pip install -e jobs
+    pip install -r jobs/requirements/default.txt -c jobs/requirements/constraints.txt
+    pushd .venv/lib/python3.6/site-packages/
     zip -r /tmp/lambda.zip *
+    popd
+    cp /tmp/lambda.zip .
     ;;
```

Comment:
This makes sure the creation of the `lambda.zip` is exclusive. It only has `buildhub` and `requirements/default.txt` in a dedicated virtualenv that exists purely for the sake of being packaged into a `.zip`.

One thing that makes me nervous about this is the fact that `buildhub` itself is not in the .zip. 

```sh
▶ aunpack -l buildhub-lambda-v1.1.5-3-g3d56a0c.zip | grep buildhub
Archive:  buildhub-lambda-v1.1.5-3-g3d56a0c.zip
       11  02-27-2018 15:52   buildhub.egg-link
```
Is that correct @leplatrem ?

Reply:
Hmm, maybe not, look at the latest release:

```
unzip -l buildhub-lambda-v1.1.5.zip | grep buildhub
Archive:  buildhub-lambda-v1.1.5.zip
        0  2018-02-23 15:34   buildhub/
     8292  2017-10-11 11:18   buildhub/to_kinto.py
    16121  2018-02-22 15:51   buildhub/inventory_to_records.py
    14247  2018-02-22 15:51   buildhub/utils.py
     6584  2017-09-14 14:24   buildhub/initialization.yml
     6953  2018-02-15 09:52   buildhub/lambda_s3_inventory.py
        0  2017-05-26 15:35   buildhub/__init__.py
[...]
```

Reply:
I think it might be because of the use of `-e`. Thanks for pointing this out.